roku_dev_cli/roku_dev_cli.py

```diff
@@ -95,14 +104,201 @@ def deployZip(zipPath, ipAddress, user):
+                    # calculate the most frequent rendezvous and durations
+                    start = data["start"]
+                    key = "%s(%i)" % (start.filePath, start.lineNumber)
+                    if key not in rByPathAndLine:
+                        rByPathAndLine[key] = {
+                            "count" : 0,
+                            "totalTime" : 0,
+                            "longestDuration" : 0
+                        }
+                    rByPathAndLine[key]["count"] += 1
+                    rByPathAndLine[key]["totalTime"] += data["end"].duration
+                    rByPathAndLine[key]["longestDuration"] = max(rByPathAndLine[key]["longestDuration"], data["end"].duration)
+
+                    # make an array of lines and mark the line with rendezvous as True
+                    filePath = start.filePath
+                    if filePath not in fileLineSections:
+                        fileLineSections[filePath] = {
+                            "rendezvousLines" : []
+                        }
+                    lineNumber = start.lineNumber
+                    rendezvousLines = fileLineSections[filePath]["rendezvousLines"]
+                    while lineNumber > len(rendezvousLines): # make sure it's big enough
+                        rendezvousLines.extend([False] * 100)
+                    rendezvousLines[lineNumber] = True
+
+            fd.write("=== Rendezvous by Frequency ===\n")
+            for pathAndLine,stats in sorted(rByPathAndLine.items(), key=lambda kv: kv[1]["count"], reverse=True):
+                averageDuration = stats["totalTime"] / stats["count"]
+                fd.write("%i - %s (avg. duration: %.3fs)\n" % (stats["count"], pathAndLine, averageDuration))
+
+            fd.write("\n")
+
+            fd.write("=== Rendezvous by Longest Single Duration ===\n")
+            for pathAndLine,stats in sorted(rByPathAndLine.items(), key=lambda kv: kv[1]["longestDuration"], reverse=True):
+                fd.write("%2.3fs - %s\n" % (stats["longestDuration"], pathAndLine))
+
+            fd.write("\n")
+
+            fd.write("=== Adjacent Rendezvous (opportunity to use `queueFields`?) ===\n")
+
+            for filePath,stats in fileLineSections.items():
+                rendezvousLines = stats["rendezvousLines"]
+
+                adjacentLines = []
+                for lineNumber in range(0, len(rendezvousLines)):
+                    if rendezvousLines[lineNumber]: # this line has a rendezvous, search adjacent lines
+                        for adjacentLineNumber in range(lineNumber-adjacentRange, lineNumber+adjacentRange+1):
+                            if adjacentLineNumber > -1 and adjacentLineNumber < len(rendezvousLines) and lineNumber != adjacentLineNumber:
+                                if rendezvousLines[adjacentLineNumber]:
+                                    adjacentLines.append(lineNumber)
+                                    break
```

Comment:
Any idea if we need to `sort -u` (or similar) the `adjacentLines` list when we're done with the `for lineNumber in …` loop?  I'm curious if we'll see skewed results because of the adjacency, especially if `adjacentRange` ever grows larger than one.  For example, something like:

* Rendezvous on line 5
* Rendezvous on line 6
* Rendezvous on line 7

would result in `adjacentLines` being `[6, 5, 7, 6]` if I'm reading this correctly?  Or maybe I'm not.  Line crunching always taking me a while to get right 🤷‍♂️

Reply:
adjacentLines should always be in order because the top for-loop goes line by line (rendezvous or not).  The inner loop looking for adjacencies also breaks when it finds a neighbor so it should only be added once.  

In your example, once the 6 is added, it should break out of the inner for-loop (looking for a neighbor) and move on to the next line.